parachain-tracer/src/tracker.rs

```diff
@@ -304,6 +308,15 @@ impl ParachainBlockTracker for SubxtTracker {
 					}
 					self.stats.on_backed();
 					metrics.on_backed(self.para_id);
+					if let Some(last_on_demand_order_block) = self.last_on_demand_order_block {
+						metrics.on_on_demand_delay(
+							relay_block_number.saturating_sub(last_on_demand_order_block),
+							self.para_id,
+						)
+					}
+					if let Some(diff) = self.get_time_diff(self.current_relay_block_ts, self.last_on_demand_order_ts) {
```

Comment:
This seems to not count the latency just for the first block. It is possible that the core might be assigned to same parachain for multiple blocks so there will be an increase in latency as blocks are built.